### DISCUSSION TITLE
Using navigationView along with UNO's extension navigator, how to update the frame of the main page

### DISCUSSION THREAD

**Discussion by digger73** [unoplatform/uno#12875] [Q&A] [answered]
Hi

I'm trying to use navigationView, using the example on[ microsofts page](https://learn.microsoft.com/en-us/windows/apps/design/controls/navigationview#code-example) along with uno's extension for navigator. However, there seems to be something I haven't figured out yet. 
If you don't use the extension methods provided by UNO, I can get the frame correctly updated however the viewModel isn't correctly "assigned" to the dataContext of the newly loaded page. Which is why I want to use the extension method. 

```
        private void NavView_Navigate(Type navPageType, NavigationTransitionInfo transitionInfo) {
            Type preNavPageType = ContentFrame.CurrentSourcePageType;

            if (navPageType is not null && !Type.Equals(preNavPageType, navPageType)) {
                 ContentFrame.Navigate(navPageType, null, transitionInfo);
            }
        }
```

However, when trying to use the extension method `NavigateViewAsync` it doesn't load into the Frame, it loads the complete page, but now the dataContext corresponds to the correct viewModel. I have tried below, and many other solutions as well, they all seems to replace the current page instead of loaded into the frame ?

`ContentFrame.Navigator()!.NavigateViewAsync(ContentFrame, navPageType);`

How do I get the NavigateViewAsync to load into the frame on the main page and not replace the main page ?

**Comment by digger73** [comment, marked as answer]
FInally figured out the connection. It is described here https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Tutorials/Navigation/Advanced/HowTo-UseNavigationView.html, unfurtenately searching on the page didn't show it as some of the first hits....

**Comment by sasakrsmanovic** [comment]
Thanks for closing the loop. I'd love to troubleshoot our search index. Would you mind to tell me, if you recall what they were, what were the search terms you used?

**Comment by GuyMicciche** [comment]
Hello, can you please help me I am having the same issue as well. Here is my project: https://github.com/unoplatform/uno/discussions/16215
